pubsub/aws/snssqs/snssqs.go

```diff
@@ -98,11 +113,28 @@ func parseInt64(input string, propertyName string) (int64, error) {
 	return int64(number), nil
 }
 
+func parseBool(input string, propertyName string) (bool, error) {
+	val, err := strconv.ParseBool(input)
+	if err != nil {
+		return false, fmt.Errorf("parsing %s failed with: %w", propertyName, err)
+	}
+	return val, nil
+}
+
 // sanitize topic/queue name to conform with:
 // https://docs.aws.amazon.com/AWSSimpleQueueService/latest/SQSDeveloperGuide/quotas-queues.html
-func nameToAWSSanitizedName(name string) string {
-	s := []byte(name)
+func nameToAWSSanitizedName(name string, isFifo bool) string {
+	suffix := ".fifo"
+
+	// first remove suffix if exists, and user requested a FIFO name, then sanitize the passed in name.
+	hasFifoSuffix := false
+	if strings.HasSuffix(name, suffix) && isFifo {
+		hasFifoSuffix = true
+		name = name[:len(name)-len(suffix)]
+	}
 
+	s := []byte(name)
+	maxLength := 80
```

Comment:
Consider declaring this as a constant up front (e.g. `maxAWSNameLength`)

Reply:
👍🏿